.github/workflows/publish.yaml

```
@@ -65,3 +65,30 @@ jobs:
           NPM_CONFIG_PROVENANCE: true
           NPM_TAG: ${{ steps.extract-tag.outputs.NPM_TAG }}
         working-directory: ${{ steps.extract-tag.outputs.NPM_PACKAGE }}
+  
+  publish-contracts-soldeer:
+    runs-on: ubuntu-latest
+    needs: publish
+    if: startsWith(github.ref_name, 'contracts/v')
+    steps:
+      - name: Extract version from tag
+        id: extract-tag-version
+        env:
+          REF_NAME: ${{ github.ref_name }}
+        run: |
+          echo "CONTRACTS_VERSION=$(echo $REF_NAME | grep -oP '(?<=v)[0-9]+\.[0-9]+\.[0-9]+')" >> $GITHUB_OUTPUT
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+      - name: Install soldeer
+        run: cargo install soldeer
+      - name: Set soldeer access token
+        env:
+          SOLDEER_ACCESS_TOKEN: ${{ secrets.SOLDEER_ACCESS_TOKEN }}
```

Let's call the env variable `SOLDEER_AUTH_TOKEN` and the secret `SOLDEER_TOKEN` for consistency with [`NPM_TOKEN`](https://github.com/oasisprotocol/sapphire-paratime/blob/main/.github/workflows/publish.yaml#L61) above. I populated the `SOLDEER_TOKEN` secret so you can test it out.

```suggestion
          SOLDEER_AUTH_TOKEN: ${{ secrets.SOLDEER_TOKEN }}
```

---

Updated.